pkg/clients/rds/rds_test.go

```diff
@@ -21,9 +48,252 @@ func TestIsErrorNotFound(t *testing.T) {
+				t.Errorf("\nIsErrorNotFound(...) %s\n", diff)
+			}
+		})
+	}
+}
+
+func TestNewClient(t *testing.T) {
+	ctx := context.TODO()
+	type args struct {
+		ctx             context.Context
+		accessKeyID     string
+		accessKeySecret string
+		securityToken   string
+		region          string
+	}
+	type want struct {
+		err error
+	}
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"FailedToNewClient": {
+			args: args{
+				ctx:             ctx,
+				accessKeyID:     "xwerwrfYfwq934tsfsFAKED",
+				accessKeySecret: "fsdfwerfaUIIffaYYYYYFUUFHUDSDKSDFAKED",
+				region:          "cn-beijing",
+			},
+			want: want{
+				err: nil,
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			_, err := NewClient(tc.args.ctx, tc.args.accessKeyID, tc.args.accessKeySecret, tc.args.securityToken, tc.args.region)
+			if diff := cmp.Diff(tc.want.err, err, test.EquateErrors()); diff != "" {
+				t.Errorf("\nNewClient(...) -want error, +got error:\n%s\n", diff)
+			}
+		})
+	}
+}
+
+func TestDescribeDBInstance(t *testing.T) {
+	c, _ := NewClient(context.TODO(), "xwerwrfYfwq934tsfsFAKED", "fsdfwerfaUIIffaYYYYYFUUFHUDSDKSDFAKED", "", "cn-beijing")
+	type args struct {
+		id string
+	}
+
+	type want struct {
+		obj     *DBInstance
+		errCode string
+	}
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"FailedToDescribeDBInstance": {
+			args: args{
+				id: "1",
+			},
+			want: want{
+				obj:     nil,
+				errCode: "InvalidAccessKeyId.NotFound",
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			db, err := c.DescribeDBInstance(tc.args.id)
+			if err != nil {
+				e, ok := err.(*errors.ServerError)
+				if ok {
+					if diff := cmp.Diff(tc.want.errCode, e.ErrorCode(), test.EquateConditions()); diff != "" {
+						t.Errorf("\nDescribeDBInstance(...) -want errorCode, +got errorCode:\n%s\n", diff)
+					}
+				}
+			}
+
+			if diff := cmp.Diff(tc.want.obj, db, test.EquateConditions()); diff != "" {
+				t.Errorf("\nDescribeDBInstance(...) %s\n", diff)
+			}
+		})
+	}
+}
+
+func TestCreateDBInstance(t *testing.T) {
+	c, _ := NewClient(context.TODO(), "xwerwrfYfwq934tsfsFAKED", "fsdfwerfaUIIffaYYYYYFUUFHUDSDKSDFAKED", "", "cn-beijing")
+	type args struct {
+		req CreateDBInstanceRequest
+	}
+
+	type want struct {
+		obj     *DBInstance
+		errCode string
+	}
+	cases := map[string]struct {
+		args args
+		want want
+	}{
+		"FailedToCreateDBInstance": {
+			args: args{
+				req: CreateDBInstanceRequest{
+					Name:                  "abc",
+					Engine:                "mysql",
+					EngineVersion:         "8.0",
+					DBInstanceClass:       "big",
+					DBInstanceStorageInGB: 20,
+					SecurityIPList:        "1.2.3.0/24",
+				},
+			},
+			want: want{
+				obj:     nil,
+				errCode: "InvalidAccessKeyId.NotFound",
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			db, err := c.CreateDBInstance(&tc.args.req)
+			if err != nil {
```

Comment:
nitpick: We might be checking whether we expect an error or not via a test case parameter, i.e., if `err == nil` but we were expecting an error, we could be failing. I think the idea here is that we are still checking the returned `db` instance but this way it would be more robust and explicit.

Reply:
As we are calling RDS Client directly with faked credentials, errors will occur definitely. So I can only check the error.